Readme.md

```diff
@@ -1,7 +1,85 @@
+The **Malware Sample Exchange (MSE)** provides a modern alternative to [Virex](https://github.com/NextSecurity/virex), to exchange malware samples between AV industry partners. It is easy to set up and supports cloud-native workflows.
+
+See the blog post for a longer introduction: [A modern Sample Exchange System](https://www.gdatasoftware.com/blog/2020/10/36410-a-modern-sample-exchange-system)
+
+Our idea is to provide a standardized exchange system which meets the following criteria:
+
+- The ability to choose only the malware samples you need. 
+- The ability for partners to filter before the download.
+  - SHA265 
+  - Categories which can include, but is not limited to, the target platform or specific detections.
+- Easy to consume API and built on current web standards ([OpenAPI](https://www.openapis.org/))
+- Easy to set up in a few minutes, so that every exchange partner is able to host it with little added effort.
+- Specific sample sets per partner
+
+## HTTP API
+
+The **Malware Sample Exchange** service is [OpenAPI](https://www.openapis.org/) compatible and exposes an API description that can be used to automatically generate a client for a programming language.
+
+HTTP API Web UI: `http://{}/swagger/index.html`
+
+HTTP API Json description: `http://{}/swagger/v1/swagger.json`
+
+Exposed End-Points:
+
+|Route|Parameter|Example|Basic Auth|Description|
+|-----|---------|-------|--------------|-----------|
+|/swagger/index.html|-|-|No|Shows the OpenAPI web interface|
+|/swagger/v1/swagger.json|-|-|No|OpenAPI json description|
+|/v1/list|start (required), end (optional)| /v1/list?start=2020-09-23 |Yes (user:password) |Fetch list with available samples
+|/v1/download|token (required)| /v1/download?token=eyJ0eX... |No |Download a sample with a token from the list
+
+## Usage
+
+For example, by executing:
+
+```bash
+# Get all samples for the user "testuser" in the time range 2020-09-23 until now
+curl -u "testuser:somenicepassword" -X GET -k -i 'http://localhost:8080/v1/list?start=2020-09-23'
+```
+
+you will receive
+a list of [JWT](https://jwt.io/) tokens, which generally have the format ```aaaa.bbbbbbbb.cccc``` of three base64 encoded sections that are separated
+by dots. The first section are header information, declaring the structure as JWT and the used hash algorithm. The second part is the
+actual payload that contains expiration date of the token, SHA256, file size and the platform. The third part is a signature that guarantees that the JWT is valid.
+If after checking that the sample is not already part of your collection and you have interest in the reported platform,
+you can download it with: 
+
+```bash
+# Download a specific sample
+curl -X GET -k "http://localhost:8080/v1/download?token=$PUT_TOKEN_HERE"
+```
+
+Not authentication is needed for the download, as the [JWT](https://jwt.io/) is signed and as such authenticates the request.
```

Comment:
s/Not authentication/No authentication/;

Reply:
fixed